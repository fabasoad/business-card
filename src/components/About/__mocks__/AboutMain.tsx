import * as React from 'react'

export default function AboutMain() {
  return (
    <div data-testid="AboutMain"></div>
  )
}
