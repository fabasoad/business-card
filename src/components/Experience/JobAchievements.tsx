import * as React from 'react'
import { useTranslation } from 'react-i18next'

interface JobAchievementsProps {
  count: number
  i18nKeyPrefix: string
}

export default function JobAchievements(props: JobAchievementsProps) {
  const { t } = useTranslation()
  return (
    <>
      <h5>{t('business-card-experience-achievements-title')}</h5>
      {Array(props.count).fill(props.i18nKeyPrefix)
        .map((p, i) => `${p}-${i + 1}`)
        .map((key) => (<p key={key}>{t(key)}</p>))}
    </>
  )
}
