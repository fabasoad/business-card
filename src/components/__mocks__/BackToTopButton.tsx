import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
