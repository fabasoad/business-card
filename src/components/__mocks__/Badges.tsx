import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
