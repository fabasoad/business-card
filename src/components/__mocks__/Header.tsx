import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
