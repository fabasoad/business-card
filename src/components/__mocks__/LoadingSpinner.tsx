import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
