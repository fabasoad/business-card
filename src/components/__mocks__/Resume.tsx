import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
