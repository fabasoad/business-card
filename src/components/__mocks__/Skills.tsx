import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
