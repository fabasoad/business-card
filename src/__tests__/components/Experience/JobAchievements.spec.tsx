/// <reference types="jest" />
import * as React from 'react'
import { shallow } from 'enzyme'
import JobAchievements from '../../../components/Experience/JobAchievements'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeEach(() => {
  tMock = useTranslation().t
})

test('should render JobAchievements correctly', () => {
  const prefix = 'some-prefix'
  const count = 2
  const wrapper = shallow(<JobAchievements
    count={count}
    i18nKeyPrefix={prefix}
  />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toBeCalledWith('business-card-experience-achievements-title')
  for (let i = 1; i < count; i++) {
    expect(tMock).toHaveBeenNthCalledWith(i + 1, `${prefix}-${i}`)
  }
})
