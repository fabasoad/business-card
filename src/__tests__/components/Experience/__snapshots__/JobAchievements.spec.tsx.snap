// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render JobAchievements correctly 1`] = `
<Fragment>
  <h5>
    business-card-experience-achievements-title
  </h5>
  <p
    key="some-prefix-1"
  >
    some-prefix-1
  </p>
  <p
    key="some-prefix-2"
  >
    some-prefix-2
  </p>
</Fragment>
`;
